### PR TITLE
Poedit: ensure scripts work no matter how many plurals a language has

### LINKED ISSUE #16318: NVDA-2024.1 enhancements to Poedit do not work
cc @LeonarddeR 
### Steps to reproduce:
1. Open a po file in Poedit3, such as NVDA’s latest interface translation.
2. Select an item in the list, for example: "Native input"
3. Try the following gestures:
  - Ctrl+Shift+C
  - Ctrl+Shift+O
  - Ctrl+Shift+W
  - Ctrl+Shift+A
### Actual behavior:
NVDA reports:
  - No comment
  - 翻译者注释
  - No translation warning
  - No notes for translators

### Expected behavior:
At least translator notes should be reported:
‪Translators: A mode  that allows typing in the 
‪actual 'native' characters for an east-Asian input 
‪method language currently selected, rather than 
‪alpha numeric (Roman/English) characters.

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA2024.1, 2024.2 Alpha
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.4170
#### Name and version of other software in use when reproducing the issue:
Poedit Free 3.4.2 (6902)
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
n/a

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #16318

### Summary of the issue:
Recently the poedit appModule was rewritten to support poedit 3.4. However, scripts for reading translation nodes, comments and old source text did not function for languages that had no plurals (Chinese) or 2 or more plurals (Polish). It did however function for languages with exactly one plural (such as French).
As the controlIDs for windows in poedit are not static, but do stay relative to each other, the appModule originally used offsets from the main dataView control. This worked okay for the translation warning, but did not work for controls in the sidebar in all cases, as it seems that there are extra controlIDs consumed for some hidden windows between the main splitter and the sidebar, depending on how many plurals a language has.
E.g. In French, the controlID offset for the Sidebar (relative to the Dataview control) is 29. In Chinese it is 27, and in Plish it is 33. I think it might be roughly two offsets per plural.
  
### Description of user facing changes
Poedit scripts such as Report translation notes (control+shift+a), Report comments (control+shift+c) and Report old source text (control+shift+o) now function no matter how many plurals a language has.

### Description of development approach
* Control ID offsets for controls in the sidebar (such as translation notes, old source text, and comments) are now relative to the sidebar itself, rather than the Dataview control.
* The sidebar window is located by first finding the main splitter window (using a controlID relative to the dataview control) and then finding the next visible sibling window from there.
* To aide in refactoring, support for controlID offsets in the Pro version are now handled by simply minusing 5 from the given controlID offset, as all the offsets in the pro version differed by 5.

### Testing strategy:
With a fr, zh_CN, and pl po files, tested control+shift+w, control+shift+a, control+shift+c and control+shift+o on an entry that contained a warning, a translation note, a comment, and old source text respectively.
* [ ] Do the same tests with the Pro version. Anyone have the pro version?

### Known issues with pull request:
None known.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by LeonarddeR** [comment]
@michaelDCurran Thanks a lot for diving into this!
@zstanecic or @ruifontes, could you have a try with the pro version?

**Comment by LeonarddeR** [comment]
Also mentioning @gregjozk, @DraganRatkovich, @Adriani90, @CyrilleB79 and @cary-rowen to get broader feedback.

**Comment by CyrilleB79** [comment]
The issue was present in 2024.1 and has been reported on mailing lists.
So a change log item should be added (bug fix).

**Comment by cary-rowen** [comment]
LGTM, I tested in Simplified Chinese environment and everything works as expected.

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -90,6 +90,7 @@ There are many minor bug fixes for applications, such as Thunderbird, Adobe Read
 * In Firefox and Chromium-based browsers, NVDA will correctly enter focus mode when pressing enter when positioned within a presentational list (ul / ol) inside editable content. (#16325)
 * Column state change is now correctly reported when selecting columns to display in Thunderbird message list. (#16323)
 * The command line switch `-h`/`--help` works properly again. (#16522, @XLTechie)
+* NVDA's support for the Poedit 3.4 translation software correctly functions when translating languages with 0 or more than 2 plurals (E.g. Chinese, Polish). (#16318)
```
Comment (seanbudd) [inline user_docs/en/changes.md:93]:
```suggestion
* NVDA's support for the Poedit 3.4+ translation software correctly functions when translating languages with 0 or more than 2 plurals (e.g. Chinese, Polish). (#16318) 
```

File: source/appModules/poedit.py [hunk 2]
```diff
@@ -79,42 +91,61 @@ def _get__dataViewControlId(self) -> int | None:
...
 
+	_sidebarControlId: int | None
+	"""Type definition for auto prop '_get__sidebarControlId'"""
+
+	def _get__sidebarControlId(self) -> int | None:
+		dataViewControlId = self._dataViewControlId
+		splitterControlID = dataViewControlId + _WindowControlIdOffsetFromDataView.MAIN_SPLITTER_IDENTIFIER
+		fg = api.getForegroundObject()
+		splitterHwnd = windowUtils.findDescendantWindow(fg.windowHandle, controlID=splitterControlID)
+		print(f"splitter: {winUser.getClassName(splitterHwnd)}")
```
Comment (seanbudd) [inline source/appModules/poedit.py:102]: is this message useful? if so convert to a log statement else remove

```suggestion
		log.debug(f"splitter: {winUser.getClassName(splitterHwnd)}")
```

```suggestion
```

File: source/appModules/poedit.py [hunk 3]
```diff
@@ -79,42 +91,61 @@ def _get__dataViewControlId(self) -> int | None:
...
+		sidebarHwnd = winUser.getWindow(splitterHwnd, winUser.GW_HWNDNEXT)
+		while sidebarHwnd and not ctypes.windll.user32.IsWindowVisible(sidebarHwnd):
+			print(f"skipping non-sidebar: {winUser.getClassName(sidebarHwnd)}")
```
Comment (seanbudd) [inline source/appModules/poedit.py:105]: same here

```suggestion
			log.debug(f"skipping non-sidebar: {winUser.getClassName(sidebarHwnd)}")
```

```suggestion
```

File: source/appModules/poedit.py [hunk 4]
```diff
@@ -79,42 +91,61 @@ def _get__dataViewControlId(self) -> int | None:
...
+			sidebarHwnd = winUser.getWindow(sidebarHwnd, winUser.GW_HWNDNEXT)
+		print(f"sidebar? {winUser.getClassName(sidebarHwnd)}")
```
Comment (seanbudd) [inline source/appModules/poedit.py:107]: I think the content of this message needs updating if it remains as a log message.
I'm going to suggest removal here unless a more descriptive message is used

```suggestion
```

**Review by seanbudd — COMMENTED** [review]
Generally looks good to me

**Review by seanbudd — APPROVED** [review]

**Pull request merged**